File: flutter_local_notifications/darwin/Classes/Converters.m
#import "Converters.h"

@implementation Converters

+ (UNNotificationCategoryOptions)parseNotificationCategoryOptions:
    (NSArray *)options {
  int result = UNNotificationCategoryOptionNone;

  for (NSNumber *option in options) {
    result |= [option intValue];
  }

  return result;
}

+ (UNNotificationActionOptions)parseNotificationActionOptions:
    (NSArray *)options {
  int result = UNNotificationActionOptionNone;

  for (NSNumber *option in options) {
    result |= [option intValue];
  }

  return result;
}

@end

File: flutter_local_notifications/macos/flutter_local_notifications/Sources/flutter_local_notifications/Converters.swift
import UserNotifications

public class Converters {
    public static func parseNotificationCategoryOptions(_ options: [Any]?) -> UNNotificationCategoryOptions {
        var result: UInt = 0

        for option in options ?? [] {
            guard let option = option as? NSNumber else {
                continue
            }
            result |= option.uintValue
        }

        return UNNotificationCategoryOptions(rawValue: result)
    }

    public static func parseNotificationActionOptions(_ options: [Any]?) -> UNNotificationActionOptions {
        var result: UInt = 0

        for option in options ?? [] {
            guard let option = option as? NSNumber else {
                continue
            }
            result |= option.uintValue
        }

        return UNNotificationActionOptions(rawValue: result)
    }
}
